requirements/requirements.txt

```diff
@@ -1,15 +1,13 @@
 click==8.1.3
 grpcio==1.46.0
-Jinja2==3.1.2
+jinja2==3.1.2
+markupsafe==2.1.1
 pluginbase==1.0.1
-protobuf==3.20.1
+protobuf==3.20.2
 psutil==5.9.0
-ruamel.yaml==0.17.21
-ruamel.yaml.clib==0.2.6
-setuptools==44.1.1
```

Comment:
This looks potentially dangerous

Reply:
Indeed. We should disable this bot (I guess it's an apache bot) at least until we're using something standard. (I think it would be nice to switch to something like poetry instead of our hand crafted makefiles).

Reply:
Oh it's actually a github bot that apache have enabled.

Reply:
Actually maybe this doesn't belong here anyway. But instead https://github.com/apache/buildstream/blob/master/pyproject.toml#L4 should declare version range of setuptools this package needs to build. So when build isolation is used, pip will just fetch compatible setuptools and use it.

Reply:
fwiw setuptools is NOT here because buildstream uses it at build-time. That should be in pyproject.toml as you mention above.

The reason setuptools is here (as a runtime requirement) is because buildtream uses `pkg_resources` around [here](https://github.com/apache/buildstream/blob/416bdd04a03d55e387d2d777cc115a295003daa2/src/buildstream/_pluginfactory/pluginoriginpip.py#L35) and `pkg_resources` module is provided by the `setuptools` package.

Reply:
Maybe we should consider using `importlib.resources` instead? It seems available since 3.7 which is the oldest version we support.

Reply:
Yes, that should be used with fallback to importlib_resources on old Python

Reply:
we'd actually need `importlib.metadata` here that has is only in the standard library 3.8 onwards. There's a pypi package named `importlib_metadata` though which we can use. But i think swapping the setuptools dependency in favor of `importlib_metadata` won't be a bad idea, even if it'll lead to a try-except like:

```python
try:
  import importlib.metadata as importlib_metadata
except ImportError:
  import importlib_metadata
...
importlib_metadata.do_something()
```